src/main/java/com/fouristhenumber/utilitiesinexcess/common/blocks/BlockDrum.java

```diff
@@ -0,0 +1,227 @@
+            if (drum.tank != null && drum.tank.getFluid() != null) {
+                player.addChatMessage(
+                    new ChatComponentText(
+                        "In drum: " + drum.tank.getFluid()
+                            .getLocalizedName() + " " + drum.tank.getFluid().amount + "L"));
+            } else {
+                player.addChatMessage(new ChatComponentText("Drum is empty"));
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public void onBlockPlacedBy(World world, int x, int y, int z, EntityLivingBase placer, ItemStack stack) {
+        super.onBlockPlacedBy(world, x, y, z, placer, stack);
+        TileEntity tile = world.getTileEntity(x, y, z);
+        if (tile instanceof TileEntityDrum drum) {
+            FluidStack fluid = ItemBlockDrum.getFluidFromStack(stack);
+            drum.setFluid(fluid);
+        }
+    }
+
+    @Override
+    public ArrayList<ItemStack> getDrops(World world, int x, int y, int z, int metadata, int fortune) {
+        // We spawn the correct stack on breakBlock(), make it not drop normally.
+        return new ArrayList<>();
+    }
+
+    @Override
+    public void breakBlock(World world, int x, int y, int z, Block block, int meta) {
+
+        ItemStack drop = new ItemStack(this, 1, meta);
+
+        TileEntity tile = world.getTileEntity(x, y, z);
+        if (tile instanceof TileEntityDrum drum) {
+            if (drum.tank.getFluid() != null) {
+                ItemBlockDrum.setFluid(
+                    drop,
+                    drum.tank.getFluid()
+                        .copy());
+            } else {
+                ItemBlockDrum.clearFluid(drop);
+            }
+        }
+
+        float f = 0.7F;
+        double dx = (double) (world.rand.nextFloat() * f) + (double) (1.0F - f) * 0.5D;
+        double dy = (double) (world.rand.nextFloat() * f) + (double) (1.0F - f) * 0.5D;
+        double dz = (double) (world.rand.nextFloat() * f) + (double) (1.0F - f) * 0.5D;
+
+        EntityItem entityItem = new EntityItem(world, x + dx, y + dy, z + dz, drop);
+        world.spawnEntityInWorld(entityItem);
+        world.removeTileEntity(x, y, z);
+        super.breakBlock(world, x, y, z, block, meta);
+    }
+
+    public static class ItemBlockDrum extends ItemBlock implements IFluidContainerItem {
+
+        public static int capacity = -1;
+
+        public ItemBlockDrum(Block block) {
+            super(block);
+            this.capacity = ((BlockDrum) block).capacity;
+        }
+
+        @Override
+        public FluidStack getFluid(ItemStack stack) {
+            return getFluidFromStack(stack);
+        }
+
+        @Override
+        public int getCapacity(ItemStack stack) {
+            return capacity;
+        }
+
+        @Override
+        public int fill(ItemStack stack, FluidStack resource, boolean doFill) {
+            if (resource == null) {
+                return 0;
+            }
+
+            FluidStack currentFluid = getFluid(stack);
```

Comment:
Can you test if this handles stackable fluid containers (like 8 GT water cells) well?

Reply:
This fill() and drain() functions take in the itemstack of the drum itself, so this shouldn't cause problems with GT cells. However, you do raise a good point on how the stackable drums work. I'll reduce the stacksize of filled and non-filled to 1